src/hardware/vga_draw.cpp

```diff
@@ -2661,40 +2661,57 @@ ImageInfo setup_drawing()
 
 		vga.draw.blocks = horiz_end;
 
+		video_mode.width = horiz_end * vga.draw.pixels_per_character;
+
+		double_width = vga.seq.clocking_mode.is_pixel_doubling &&
+		               vga.draw.pixel_doubling_allowed;
+
```

Comment:
I recommend checking out the file and comparing the old and new `M_TEXT` branch side by side. It's very hard to see what's going on by looking at this diff, even for me...